opentracing-impl/src/main/java/io/opentracing/propagation/TextMapImpl.java

```diff
@@ -16,7 +16,7 @@
 import java.util.Iterator;
 import java.util.Map;
 
-public final class TextMapImpl implements TextMapWriter, TextMapReader {
+public final class TextMapImpl implements TextMap {
```

Comment:
Okay, done. I went with `TextMapInjectAdapter` and `TextMapExtractAdapter` which is clearer and more self-documenting (at least in my brain).